src/App.tsx

```diff
@@ -1,25 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const getPageItems = (
+  total: number,
+  itemsPerPage: number,
+  currentPage: number,
+) => {
+  return [...getNumbers(1, total)].slice(
+    itemsPerPage * (currentPage - 1),
+    itemsPerPage * currentPage,
+  );
+};
+
+const ITEMS_COUNT = 42;
+const PER_PAGE_COUNT = 5;
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(PER_PAGE_COUNT);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const pageItems = getPageItems(ITEMS_COUNT, perPage, currentPage);
+
+  const handlePerPageSelectorChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setPerPage(+event.target.value);
+    setCurrentPage(1);
+  };
+
+  const handlePageChange = (page: number) => {
+    setCurrentPage(page);
+  };
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        Page {currentPage} (items {pageItems[0]} -{' '}
+        {pageItems[pageItems.length - 1]} of {ITEMS_COUNT})
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
             data-cy="perPageSelector"
             id="perPageSelector"
-            className="form-control">
+            className="form-control"
+            onChange={handlePerPageSelectorChange}
+            defaultValue={PER_PAGE_COUNT}
```

Comment:
Using defaultValue on the select element is not recommended when the value is controlled by React state. You should use the 'value' attribute instead.